addons/mail/static/src/core/common/channel_member_model.js

```diff
@@ -35,6 +34,25 @@ export class ChannelMember extends Record {
     });
     fetched_message_id = Record.one("Message");
     seen_message_id = Record.one("Message");
+    syncUnread = true;
+    syncUnreadInNeed = Record.attr(false, {
+        compute() {
+            if (!this.syncUnread || !this.eq(this.thread?.selfMember)) {
+                return false;
+            }
+            return (
+                this.localNewMessageSeparator !== this.new_message_separator ||
+                this.localMessageUnreadCounter !== this.thread.message_unread_counter
+            );
+        },
+        onUpdate() {
+            if (this.syncUnreadInNeed) {
+                this.localNewMessageSeparator = this.new_message_separator;
+                this.localMessageUnreadCounter = this.thread.message_unread_counter;
+            }
+        },
+    });
```

Comment:
We could rename `syncUnreadInNeed` to `_syncUnread`, so it's very clear it's internal code for explicit usage of `syncUnread`.

Probably would be nice to avoid making another field, that this kind of feature is built-in in models and you just need to define a single field for it to allow write and custom-self-side-effect